packages/vue/_src/hooks.ts

```diff
@@ -206,24 +206,50 @@ export function make<R, E, A>(self: Effect<R, E, FetchResponse<A>>) {
   return tuple(result, latestSuccess, execute)
 }
 
+export type MutationResult<E, A, I = never> = readonly [
+  {
+    loading: Ref<boolean>
+    success: Ref<A | undefined>
+    error: Ref<E | undefined>
+  },
+  (
+    ...args: [I] extends [never] ? [abortSignal?: AbortSignal] : [i: I, abortSignal?: AbortSignal]
+  ) => Promise<Either<E, A>>
+]
+
 /**
- * Pass a function that returns an Effect, e.g from a client action.
+ * Pass a function that returns an Effect, e.g from a client action, or an Effect
  * Returns a tuple with state ref and execution function which reports errors as Toast.
  */
-export function useMutation<I, E, A>(self: (i: I) => Effect<ApiConfig | Http, E, A>) {
+export const useMutation: {
```

Comment:
There is still just one `exec` to avoid duplicating the Effect pipeline, but the overall situation is cleaner now